src/rgw/rgw_bl.cc

```diff
@@ -0,0 +1,984 @@
+  std::string newliner = "\n";
+  std::stringstream pending_column;
+  std::string oname;
+  std::string oversion_id;
+
+  oname = entry.obj.name.empty() ? "-" : entry.obj.name;
+  oversion_id = entry.obj.instance.empty() ? "-" : entry.obj.instance;
+
+  struct tm entry_time;
+  time_t tt = entry.time.sec();
+  localtime_r(&tt, &entry_time);
+  char time_buffer[29];
+  strftime(time_buffer, 29, "[%d/%b/%Y:%H:%M:%S %z]", &entry_time);
+  std::string time(time_buffer);
+
+  std::string owner_id = entry.bucket_owner.id.empty() ? "-" : entry.bucket_owner.id;
+  std::string bucket = entry.bucket.empty() ? "-" : entry.bucket;
+  std::string user = entry.user.empty() ? "-" : entry.user;
+  std::string uri = entry.uri.empty() ? "\"-\"" : ("\""+entry.uri+"\"");
+  std::string error_code = entry.error_code.empty() ? "-" : entry.error_code;
+  std::string referrer = entry.referrer.empty() ? "\"-\"" : ("\""+entry.referrer+"\"");
+  std::string user_agent = entry.user_agent.empty() ? "\"-\"" : ("\""+entry.user_agent+"\"");
+
+                                                                               // S3 BL field
+  pending_column << owner_id << row_separator                                  // Bucket Owner
+                 << bucket << row_separator                                    // Bucket
+                 << time << row_separator                                      // Time
+                 << entry.remote_addr << row_separator                         // Remote IP
+                 << user << row_separator                                      // Requester
+                 << entry.request_id << row_separator                          // Request ID
+                 << entry.op << row_separator                                  // Operation
+                 << oname << row_separator                                     // Key
+                 << uri << row_separator                                       // Request-URI
+                 << entry.http_status << row_separator                         // HTTP status
+                 << error_code << row_separator                                // Error Code
+                 << entry.bytes_sent << row_separator                          // Bytes Sent
+                 << entry.obj_size << row_separator                            // Object Size
+                 << entry.total_time.to_msec() << row_separator                // Total Time
+                 << "-" << row_separator                                       // Turn-Around Time
+                 << referrer << row_separator                                  // Referrer
+                 << user_agent << row_separator                                // User-Agent
+                 << oversion_id << row_separator                               // Version Id
+                 << newliner;
+
+  buffer->append(pending_column.str());
+}
+
+int RGWBL::bucket_bl_upload(bufferlist* opslog_buffer, rgw_obj obj,
+			    map<string, bufferlist> tobject_attrs)
```

Comment:
Need to rewrite this method to create an object directly instead of sending a message to ourselves